pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -881,11 +881,15 @@ func podMatchesNodeSelectorAndAffinityTerms(pod *v1.Pod, node *v1.Node) bool {
 
 		// Match node selector for requiredDuringSchedulingIgnoredDuringExecution.
 		if nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
-			nodeSelectorTerms := nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms
-			klog.V(10).Infof("Match for RequiredDuringSchedulingIgnoredDuringExecution node selector terms %+v", nodeSelectorTerms)
-			nodeAffinityMatches = nodeAffinityMatches && nodeMatchesNodeSelectorTerms(node, nodeSelectorTerms)
+			// If pod.Status.Phase is running, which means that this pod was previously scheduled and started running,
+			// the reason running this check is that kubelet is restarted and reruns canAdmitPod method,
+			// in this case, we will ignore NodeAffinity check to follow the *IgnoredDuringExectution* semantic.
+			if pod.Status.Phase != v1.PodRunning {
```

Comment:
How about `if nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil && len(pod.Spec.NodeName) != 0 {`?

Reply:
if someone explicitly assign a pod to a node  by filling in `pod.Spec.NodeName`, in this scenario, the kubelet will allow pod althought this node's label doesn't meet pod's affinity constraint.

Reply:
After revisiting this issue, I changed as you comment which seems to be more clear. 

> if someone explicitly assign a pod to a node by filling in `pod.Spec.NodeName`, in this scenario, the kubelet will allow pod although this node's label doesn't meet pod's affinity constraint.
This is a common issue and is not the goal of this PR.